From 83408c9fbdf9b41b7d40e7a6b1bdfe85c67d9beb Mon Sep 17 00:00:00 2001
From: Orlando Vazquez <ovazquez@gmail.com>
Date: Thu, 7 Jun 2018 16:00:13 -0700
Subject: [PATCH] TRITON-479 CNAPI does not allow setting of
 `overprovision_ratios` as a result of request id not being passed to
 server.sendTaskRequest()

---
 lib/endpoints/servers.js | 1 +
 lib/models/server.js     | 1 +
 package.json             | 2 +-
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/endpoints/servers.js b/lib/endpoints/servers.js
index 26f5070..8d3caa6 100644
--- a/lib/endpoints/servers.js
+++ b/lib/endpoints/servers.js
@@ -499,6 +499,7 @@ Server.update = function handlerServerUpdate(req, res, next) {
                     synccb: function (err, result) {
                         cb(err, result);
                     },
+                    req_id: req.getId(),
                     params: { value: req.params.overprovision_ratios  }
                 };
 
diff --git a/lib/models/server.js b/lib/models/server.js
index 76377cc..9853448 100644
--- a/lib/models/server.js
+++ b/lib/models/server.js
@@ -2107,6 +2107,7 @@ function (opts) {
 
     if (opts.req_id) {
         req_id = opts.req_id;
+        assert.string(req_id, 'req_id');
     } else if (opts.req) {
         req_id = opts.req.getId();
     }
diff --git a/package.json b/package.json
index ce250f5..6438f03 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.9.0",
+  "version": "1.9.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.21.0

